lib/iris/tests/__init__.py

```diff
@@ -441,10 +449,21 @@ def assertRepr(self, obj, reference_filename):
         self.assertString(repr(obj), reference_filename)
 
     def _check_same(self, item, reference_path, type_comparison_name='CML'):
+        def _patch_history_attr(cdl):
+            result = iso8601.search(cdl)
+            if result:
```

Comment:
What if there's more than one result? Can that happen? If it can, we should recursively call this until they're all gone...